drivers: spi: spi-dw cleanup and serial target binding

XenuIsWatching opened the pull request:
This cleans up the instantiation macro. This also adds a binding if the IP is configured as a serial target and fixes return of the rx_len when it is a serial target. DBG_COUNTER was also removed as that appears to be unnecessary.

aux-reg should be defined with a hyphen in the bindings

The HAS_SPI_DW Kconfig is rather unnecessary. If the synopsys designware spi is to be included. It should come from the devicetree.

XenuIsWatching committed:
drivers: spi: dw: remove HAS_SPI_DW Kconfig

The HAS_SPI_DW Kconfig is rather unncessary. If the synopsys designware
spi is to be included. It should come from the devicetree.

Signed-off-by: Ryan McClelland <ryanmcclelland@meta.com>

XenuIsWatching committed:
drivers: spi: dw: fix naming convention of aux-reg

aux-reg should be defined with a hyphen in the bindings

Signed-off-by: Ryan McClelland <ryanmcclelland@meta.com>

tbursztyka commented on drivers/spi/spi_dw.c:
```diff
@@ -439,12 +421,26 @@ static int transceive(const struct device *dev,
+		ret = spi->ctx.recv_frames;
+	}
+#endif /* CONFIG_SPI_SLAVE */
+
```
Could you split this commit in 2? one for the init macro, one for the slave fixes.

XenuIsWatching replied:
The slave fixes are also within the init macro. It's kinda difficult to split them up

XenuIsWatching replied:
done

XenuIsWatching committed:
drivers: spi: dw: cleanup instantiation macro

This cleans up the instantiation macro. DBG_COUNTER was also removed
as that appears to be unnecessary. This also allows for if it is a
serial target to be configured from the devicetree.

Signed-off-by: Ryan McClelland <ryanmcclelland@meta.com>

XenuIsWatching committed:
drivers: spi: dw: fix hw cs and slave return rx len

Only toggle the hw cs if the cs is not set as a gpio. SPI trancieve
should also return the rx len when in slave configuration.

Signed-off-by: Ryan McClelland <ryanmcclelland@meta.com>

XenuIsWatching committed:
drivers: spi: dw: define max-xfer-size

The max size was determined by looking at the ARCH of the cpu. This really
comes from the ip configuration when generated. Add `max-xfer-size`
property to the devicetree.

Signed-off-by: Ryan McClelland <ryanmcclelland@meta.com>

XenuIsWatching commented:
@tbursztyka An additional issue was found, where the spi-max-xfer-size was defined by a kConfig which was looking at the ARCH of the cpu. This really came from the configuration that the IP was generated with. This adds a devicetree property for it.

XenuIsWatching commented:
pinging @tbursztyka for review
